client/config.go

```diff
@@ -14,18 +17,55 @@ func GetConfigSystem(d *schema.ResourceData) models.ConfigBodyPost {
 }
 
 func GetConfigAuth(d *schema.ResourceData) models.ConfigBodyPost {
-	return models.ConfigBodyPost{
-		AuthMode:         d.Get("auth_mode").(string),
-		OidcName:         d.Get("oidc_name").(string),
-		OidcEndpoint:     d.Get("oidc_endpoint").(string),
-		OidcClientID:     d.Get("oidc_client_id").(string),
-		OidcClientSecret: d.Get("oidc_client_secret").(string),
-		OidcGroupsClaim:  d.Get("oidc_groups_claim").(string),
-		OidcScope:        d.Get("oidc_scope").(string),
-		OidcVerifyCert:   d.Get("oidc_verify_cert").(bool),
-		OidcAutoOnboard:  d.Get("oidc_auto_onboard").(bool),
-		OidcUserClaim:    d.Get("oidc_user_claim").(string),
+	var body models.ConfigBodyPost
+
+	auth := d.Get("auth_mode").(string)
+
+	switch auth {
+	case "oidc_auth", "oidc":
+		body = models.ConfigBodyPost{
+			AuthMode:         "oidc_auth",
+			OidcName:         d.Get("oidc_name").(string),
+			OidcEndpoint:     d.Get("oidc_endpoint").(string),
+			OidcClientID:     d.Get("oidc_client_id").(string),
+			OidcClientSecret: d.Get("oidc_client_secret").(string),
+			OidcGroupsClaim:  d.Get("oidc_groups_claim").(string),
+			OidcScope:        d.Get("oidc_scope").(string),
+			OidcVerifyCert:   d.Get("oidc_verify_cert").(bool),
+			OidcAutoOnboard:  d.Get("oidc_auto_onboard").(bool),
+			OidcUserClaim:    d.Get("oidc_user_claim").(string),
+		}
+	case "ldap_auth", "ldap":
+		body = models.ConfigBodyPost{
+			AuthMode:     "ldap_auth",
+			LdapURL:      d.Get("ldap_url").(string),
+			LdapSearchDn: d.Get("ldap_search_dn").(string),
+			LdapBaseDn:   d.Get("ldap_base_dn").(string),
+			LdapFilter:   d.Get("ldap_filter").(string),
+			LdapUID:      d.Get("ldap_uid").(string),
+
+			LdapGroupBaseDn:        d.Get("ldap_group_base_dn").(string),
+			LdapGroupSearchFilter:  d.Get("ldap_group_filter").(string),
+			LdapGroupGID:           d.Get("ldap_group_gid").(string),
+			LdapGroupAdminDn:       d.Get("ldap_group_admin_dn").(string),
+			LdapGroupAttributeName: d.Get("ldap_group_membership").(string),
+
+			LdapVerifyCert: d.Get("ldap_verify_cert").(bool),
+		}
+
+		ldapScope := d.Get("ldap_scope").(string)
+		if ldapScope != "" {
+			body.LdapScope = getLdapScope(ldapScope)
+		}
+
+		ldapGroupScope := d.Get("ldap_group_scope").(string)
+		if ldapGroupScope != "" {
+			body.LdapGroupSearchScope = getLdapScope(ldapGroupScope)
+		}
+
 	}
+	log.Printf("[DEBUG] %+v\n ", body)
```

Comment:
Maybe remove debug prints, or use something like zero to only display these print when running terraform with debug logging turned on.